### PR TITLE
Add error attribute to metrics labels

### PR DESCRIPTION
**Pull request by jedrivisser**
This PR allows you to add specific information about an error to a metric when it occurs. For instance the mysql error code.

An example implementation of the `ErrorAttributesGetter` function for mysql:

```
// For information on MySQL errors and mappings, see https://dev.mysql.com/doc/mysql-errors/5.7/en/server-error-reference.html
func MySQLErrorAttributesGetter(err error) []attribute.KeyValue {
	var mysqlError *mysql.MySQLError
	if errors.As(err, &mysqlError) {
		return []attribute.KeyValue{
			attribute.String("error_type", "mysql"),
			attribute.String("error_number", strconv.FormatUint(uint64(mysqlError.Number), 10)),
		}
	}

	return nil
}
```

This would be registered with:

```
driverName, err := otelsql.Register("mysql",otelsql.WithErrorAttributesGetter(MySQLErrorAttributesGetter))
```

It creates a metric that looks something like:

```
db_sql_latency_milliseconds_bucket{error_number="1062",error_type="mysql",method="sql.stmt.exec",otel_scope_name="github.com/XSAM/otelsql",otel_scope_version="0.36.0",status="error",le="100"} 1
```

instead of:

```
db_sql_latency_milliseconds_bucket{method="sql.stmt.exec",otel_scope_name="github.com/XSAM/otelsql",otel_scope_version="0.36.0",status="error",le="100"} 1
```

This allows us to use the error number to decide if an error is a critical error or one that is already being managed in code.

### CODE REVIEW

**Comment by jedrivisser** [comment]
Added some more unittests

File: option_test.go [hunk 1]
```diff
@@ -33,6 +33,10 @@ func TestOptions(t *testing.T) {
 		return []attribute.KeyValue{attribute.String("foo", "bar")}
 	}
 
+	dummyErrorAttributesGetter := func(_ error) []attribute.KeyValue {
+		return []attribute.KeyValue{defaultattribute}
```
Comment (XSAM) [inline option_test.go:37]: Could you use attributes that can only come from `dummyErrorAttributesGetter`, like `foo2:bar2`, instead of re-using `defaultattribute`?

This makes sure attributes in the output are actually coming from `dummyErrorAttributesGetter` instead of another part of the code.

File: utils_test.go [hunk 2]
```diff
@@ -260,6 +260,14 @@ var keep SpanFilter = func(_ context.Context, _ Method, _ string, _ []driver.Nam
 }
 
 func TestRecordMetric(t *testing.T) {
+	dummyAttributesGetter := func(_ context.Context, _ Method, _ string, _ []driver.NamedValue) []attribute.KeyValue {
+		return []attribute.KeyValue{defaultattribute}
```
Comment (XSAM) [inline utils_test.go:264]: Ditto. Could we have other attributes?

File: utils_test.go [hunk 3]
```diff
@@ -313,6 +322,28 @@ func TestRecordMetric(t *testing.T) {
...
+				cfg:    newConfig(WithInstrumentAttributesGetter(dummyAttributesGetter)),
+				method: MethodConnQuery,
+				query:  "example query",
+			},
+			recordErr:      nil,
+			expectedStatus: "ok",
+			expectedAttr:   defaultattribute,
+		},
+		{
+			name: "metric with errorAttributesGetter",
```
Comment (XSAM) [inline utils_test.go:337]: Could we add a test that uses `WithErrorAttributesGetter`, and the `recordErr` is nil?

File: utils_test.go [hunk 4]
```diff
@@ -272,6 +280,7 @@ func TestRecordMetric(t *testing.T) {
 		args           args
 		recordErr      error
 		expectedStatus string
+		expectedAttr   attribute.KeyValue
```
Comment (XSAM) [inline utils_test.go:283]: I prefer we compare the entire attribute set directly.
```suggestion
		expectedAttrs   attribute.Set
```

Then, construct and compare attributes directly

```go
expectedAttrs: attribute.NewSet(attribute.String("method", "sql.conn.query"), attribute.String("status", "ok")
```

```go
assert.Equal(t, tt.expectedAttrs, mockLatency.attrs)
```

**Review by XSAM — COMMENTED** [review]
@jedrivisser, thanks for creating this PR.

This feature seems reasonable, but some comments need to be resolved before merging.

Please also update the `CHANGELOG.me` file.

**Comment by jedrivisser** [comment]
Thanks for the review! I think I have addressed all issues. Please let me know if I missed or misunderstood anything

**Review by XSAM — APPROVED** [review]
LGTM. Thanks for your contribution!

**Comment by jedrivisser** [comment]
Awesome, thanks!
